### DISCUSSION TITLE
python3 "import Ice" failed on Linux (Docker container)

### DISCUSSION THREAD

**Discussion by eklunko** [zeroc-ice/ice#1582] [General]
### Version

Ice version:
zeroc-ice-all-runtime: 3.7.8-2.1
zeroc-ice-all-dev: 3.7.8-2.1

OS version: Docker image python3.12:bookworm

### Describe the bug

Steps to reproduce:

Run the container:
```
someuser@some-wsl-host:~$ docker run -it -h docker1 python:3.12-bookworm bash

root@docker1:/# apt update && apt upgrade -y
```

Install ICE according to last box under "Installing Ice on Debian" section on this page:
https://doc.zeroc.com/ice/3.7/release-notes/using-the-linux-binary-distributions
```
root@docker1:/# apt-get install -y zeroc-ice-all-runtime zeroc-ice-all-dev
```

Run the command and check the result:
```
root@docker1:/# python3 -c "import Ice"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'Ice'
```

### Expected behavior

"import Ice" should run without errors.

**Comment by pepone** [comment]
This is probably because we build Python packages with the default Python version for Bookworm, (Python 3.11)

The Ice package is not available with 3.12
```
root@docker1:/# python3 -c "import Ice"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'Ice'
```

But works fine with 3.11, which is the version we build for

```
root@docker1:/# python3.11 -c "import Ice"
root@docker1:/#
```

**Comment by eklunko** [comment]
> This is probably because we build Python packages with the default Python version for Bookworm, (Python 3.11)
> The Ice package is not available with 3.12
...
> But works fine with 3.11, which is the version we build for

Thank you, @pepone!
I repeated my test using `python:3.11-bookworm` Docker image but unfortunately with the same result:
```
root@docker1:/# python3.11 -c "import Ice"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'Ice'
```
Very strange! Do you use exactly the same sequence of commands as I wrote in the first message? Could it be that ICE requires some post-install configuration that I missed?

**Comment by pepone** [comment]
Yes, I did use the same sequence of commands:

```
root@docker1:/# history
    1  apt update && apt upgrade -y
    2  apt-get install -y zeroc-ice-all-runtime zeroc-ice-all-dev
    3  python3 -c "import Ice"
    4  python3.11 -c "import Ice"
    5  python3.11
```

Looking at the installed packages I see:

```
root@docker1:/# apt list --installed | grep zeroc

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

libzeroc-ice-dev/stable,now 3.7.8-2.1+b1 amd64 [installed,automatic]
libzeroc-ice3.7-java/stable,now 3.7.8-2.1 all [installed,automatic]
libzeroc-ice3.7/stable,now 3.7.8-2.1+b1 amd64 [installed,automatic]
libzeroc-icestorm3.7/stable,now 3.7.8-2.1+b1 amd64 [installed,automatic]
php-zeroc-ice/stable,now 3.7.8-2.1+b1 amd64 [installed,automatic]
python3-zeroc-ice/stable,now 3.7.8-2.1+b1 amd64 [installed,automatic]
zeroc-glacier2/stable,now 3.7.8-2.1+b1 amd64 [installed,automatic]
zeroc-ice-all-dev/stable,now 3.7.8-2.1 all [installed]
zeroc-ice-all-runtime/stable,now 3.7.8-2.1 all [installed]
zeroc-ice-compilers/stable,now 3.7.8-2.1+b1 amd64 [installed,automatic]
zeroc-ice-slice/stable,now 3.7.8-2.1 all [installed,automatic]
zeroc-ice-utils/stable,now 3.7.8-2.1+b1 amd64 [installed,automatic]
zeroc-icebox/stable,now 3.7.8-2.1+b1 amd64 [installed,automatic]
zeroc-icebridge/stable,now 3.7.8-2.1+b1 amd64 [installed,automatic]
zeroc-icegrid/stable,now 3.7.8-2.1+b1 amd64 [installed,automatic]
zeroc-icegridgui/stable,now 3.7.8-2.1 all [installed,automatic]
zeroc-icepatch2/stable,now 3.7.8-2.1+b1 amd64 [installed,automatic]
```

And the extension is installed in

```
/usr/lib/python3/dist-packages/Ice/__init__.py
/usr/lib/python3/dist-packages/IcePy.cpython-311-x86_64-linux-gnu.so
...
```

**Comment by eklunko** [comment]
Yes, my list of installed packages is exactly the same.

I also ran my test on the Linux server running Ubuntu 20.04.6 LTS and Docker version 24.0.6. With the same error.

My previous tests were on Windows 10, Docker Desktop v4.25.2, WSL backend, Docker version 24.0.6.

I don't have any ideas yet.

**Comment by pepone** [comment]
I test with Docker on Windows 10 too

```
C:\Users\jose\source\repos\docker-images>docker --version
Docker version 20.10.16, build aa7e414
```
